.azure/test-linux.yml

```diff
@@ -0,0 +1,162 @@
+parameters:
+  - name: "pythonVersion"
+    type: string
+    displayName: "Version of Python to test"
+
+  - name: "testQPY"
+    type: boolean
+    default: false
+
+  - name: "testImages"
+    type: boolean
+    default: false
+
+  - name: "installFromSdist"
+    type: boolean
+    default: false
+
+
+jobs:
+  - job: "Linux_Tests_Python${{ replace(parameters.pythonVersion, '.', '') }}"
+    displayName: "Test Linux Python ${{ parameters.pythonVersion }}"
+    pool: {vmImage: 'ubuntu-latest'}
+
+    variables:
+      QISKIT_SUPPRESS_PACKAGING_WARNINGS: Y
+      PIP_CACHE_DIR: $(Pipeline.Workspace)/.pip
+      QISKIT_TEST_CAPTURE_STREAMS: 1
+
+    steps:
+      - task: UsePythonVersion@0
+        inputs:
+          versionSpec: '${{ parameters.pythonVersion }}'
+        displayName: 'Use Python ${{ parameters.pythonVersion }}'
+
+      - task: Cache@2
+        inputs:
+          key: 'pip | "$(Agent.OS)" | "${{ parameters.pythonVersion }}" |"$(Build.BuildNumber)"'
+          restoreKeys: |
+            pip | "$(Agent.OS)" | "${{ parameters.pythonVersion }}"
+            pip | "$(Agent.OS)"
+            pip
+          path: $(PIP_CACHE_DIR)
+        displayName: "Cache pip"
+
+      - task: Cache@2
+        inputs:
+          key: 'stestr | "$(Agent.OS)" | "${{ parameters.pythonVersion }}" | "$(Build.BuildNumber)"'
+          restoreKeys: |
+            stestr | "$(Agent.OS)" | "${{ parameters.pythonVersion }}"
+            stestr | "$(Agent.OS)"
+            stestr
+          path: .stestr
+        displayName: "Cache stestr"
+
+      - bash: |
+          set -e
+          python -m pip install --upgrade pip setuptools wheel virtualenv
+          virtualenv test-job
+        displayName: "Prepare venv"
+
+      - ${{ if eq(parameters.installFromSdist, true) }}:
+        - bash: |
+            set -e
+            source test-job/bin/activate
+            pip install -U -r requirements.txt -r requirements-dev.txt -c constraints.txt
+            python setup.py sdist
+            pip install -U -c constraints.txt dist/qiskit-terra*.tar.gz
+          displayName: "Install Terra from sdist"
+
+      - ${{ if eq(parameters.installFromSdist, false) }}:
+        - bash: |
+            set -e
+            source test-job/bin/activate
+            pip install -U -r requirements.txt -r requirements-dev.txt -c constraints.txt
+            pip install -U -c constraints.txt -e .
+          displayName: "Install Terra directly"
+
+      - bash: |
+          set -e
+          source test-job/bin/activate
+          pip install -U "cplex" "qiskit-aer" "z3-solver" -c constraints.txt
+          mkdir -p /tmp/terra-tests
+          cp -r test /tmp/terra-tests/.
+          cp tools/verify_parallel_map.py /tmp/terra-tests/.
+          cp .stestr.conf /tmp/terra-tests/.
+          cp -r .stestr /tmp/terra-tests/. || :
```

Comment:
FWIW, these steps are only really needed in the install from sdist step. When we build and install with `pip install -e .` we can run tests from the repo root because the extension is built inplace. For sdist I did this more to avoid confusing python when we import qiskit because it'll try to use the repo instead of the package we installed from sdist because `qiskit` is in the cwd. But that being said it doesn't hurt anything to do this and it keeps the logic here a bit simpler.